test/over_react/component/suspense_component_test.dart

```diff
@@ -0,0 +1,143 @@
+        () async {
+          await lazy_load_me.loadLibrary();
+          await Future.delayed(Duration(seconds: 1));
+          return lazy_load_me.LazyLoadMe;
+        },
+        UiFactoryConfig(
+          displayName: 'LazyLoadMe',
+          propsFactory: PropsFactory.fromUiFactory(LazyLoadMePropsMapView),
+        ),
+      );
+
+      final view = render(
+        (Suspense()..fallback = (Dom.span()..addTestId(loadingId))())(
+          (LazyLoadMe()
+            ..addTestId(lazyId)
+            ..initialCount = 2)(),
+        ),
+      );
+
+      expect(view.getByTestId(loadingId), isNotNull);
+      expect(view.queryByTestId(lazyId), isNull);
+
+      expect(view.findByTestId(lazyId), isNotNull);
+    });
+
+    test('is instant after the lazy component has been loaded once', () async {
+      const lazyLoadArtificialDelayInSeconds = 2;
+      final oneTenthOfArtificialDelayInMilliseconds = ((lazyLoadArtificialDelayInSeconds / 10) * 1000).round();
+
+      final LazyLoadMe = lazy(
+        () async {
+          await lazy_load_me.loadLibrary();
+          await Future.delayed(Duration(seconds: lazyLoadArtificialDelayInSeconds));
+          return lazy_load_me.LazyLoadMe;
+        },
+        UiFactoryConfig(
+          displayName: 'LazyLoadMe',
+          propsFactory: PropsFactory.fromUiFactory(LazyLoadMePropsMapView),
+        ),
+      );
+
+      var mountPoint1 = DivElement();
+      var mountPoint2 = DivElement();
+
+      final view1 = render(
+          (Suspense()..fallback = (Dom.span()..addTestId(loadingId))())(
+            (LazyLoadMe()
+              ..addTestId(lazyId)
+              ..initialCount = 2)(),
+          ),
+          baseElement: mountPoint1);
+
+      expect(view1.getByTestId(loadingId), isNotNull);
+      expect(view1.queryByTestId(lazyId), isNull);
+
+      expect(view1.findByTestId(lazyId, timeout: Duration(seconds: lazyLoadArtificialDelayInSeconds + 1)), isNotNull);
+
+      final view2 = render(
+          (Suspense()..fallback = (Dom.span()..addTestId(loadingId))())(
+            (LazyLoadMe()
+              ..addTestId(lazyId)
+              ..initialCount = 2)(),
+          ),
+          baseElement: mountPoint2);
+
+      // Using a `findBy` with a timeout that is one tenth of the expected delay.
+      // This asserts that it renders almost immediately. T
+      // The loading ui will still show but is replaced almost immediately (like 1 animation frame).
+
+      expect(view2.findByTestId(lazyId, timeout: Duration(milliseconds: oneTenthOfArtificialDelayInMilliseconds)), isNotNull);
+
+      // Just asserting that `oneTenthOfArtificialDelayInMilliseconds` is in fact less than `lazyLoadArtificialDelayInSeconds`.
+      expect(oneTenthOfArtificialDelayInMilliseconds, 200);
+      expect(
+          Duration(milliseconds: oneTenthOfArtificialDelayInMilliseconds).inMilliseconds <
+              Duration(seconds: lazyLoadArtificialDelayInSeconds).inMilliseconds,
+          isTrue);
```

Comment:
#nit I think we could reduce complexity slightly, and reduce potential CI flakiness by using a completer for the artificial delay:
```dart
    test('is instant after the lazy component has been loaded once', () async {
      final lazyLoadCompleter = Completer<void>();
      const suspenseFallbackTimeout = Duration(milliseconds: 100);

      final LazyLoadMe = lazy(
        () async {
          await lazy_load_me.loadLibrary();
          await lazyLoadCompleter.future;
          return lazy_load_me.LazyLoadMe;
        },
        UiFactoryConfig(
          displayName: 'LazyLoadMe',
          propsFactory: PropsFactory.fromUiFactory(LazyLoadMePropsMapView),
        ),
      );

      {
        final view = render(
          (Suspense()..fallback = (Dom.span()..addTestId(loadingId))())(
            (LazyLoadMe()
              ..addTestId(lazyId)
              ..initialCount = 2
            )(),
          ),
        );

        expect(view.getByTestId(loadingId), isNotNull);
        expect(view.queryByTestId(lazyId), isNull);

        await Future.delayed(Duration(seconds: 1));
        expect(view.queryByTestId(lazyId), isNull, reason: 'should not be found until we the lazy component finishes loading');

        lazyLoadCompleter.complete();
        expect(view.findByTestId(lazyId, timeout: suspenseFallbackTimeout),
            isNotNull);
        view.unmount();
      }
      {
        final view = render(
          (Suspense()..fallback = (Dom.span()..addTestId(loadingId))())(
            (LazyLoadMe()
              ..addTestId(lazyId)
              ..initialCount = 2
            )(),
          ),
        );
        // This asserts that it renders almost immediately.
        // The loading ui will still show but is replaced almost immediately (like 1 animation frame).
        expect(view.findByTestId(lazyId, timeout: suspenseFallbackTimeout), isNotNull);
      }
    });
```